sycl/source/detail/queue_impl.hpp

```diff
@@ -166,20 +168,22 @@ class queue_impl {
   /// \param Self is a shared_ptr to this queue.
   /// \param SecondQueue is a shared_ptr to the secondary queue.
   /// \param Loc is the code location of the submit call (default argument)
+  /// \param StoreAdditionalInfo makes additional info be stored in event_impl
   /// \return a SYCL event object, which corresponds to the queue the command
   /// group is being enqueued on.
-  event submit(const function_class<void(handler &)> &CGF, bool *IsKernel,
+  event submit(const function_class<void(handler &)> &CGF,
                const shared_ptr_class<queue_impl> &Self,
                const shared_ptr_class<queue_impl> &SecondQueue,
-               const detail::code_location &Loc) {
+               const detail::code_location &Loc,
+               const SubmitPostProcessF *PostProcess = nullptr) {
```

Comment:
Can we have an empty std::function instead of using pointers here?

Reply:
This was an option at first, but I'd like to not introduce another object copy here.